### PR TITLE
[Xamarin.Android] Xamarin.Android pre-release changes

### PR DESCRIPTION
**Pull request by almedina-ms**
## Related Issue
#3626

## Description
Adds the changes for Xamarin.Android bindings and first version of a sample application

## How Verified
This was all tested manually using the sample application for doing this binding

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/3610)

### CODE REVIEW

File: source/android/README.md [hunk 1]
```diff
@@ -25,6 +25,12 @@ swig -c++ -java -package io.adaptivecards.objectmodel -outdir main/java/io/adapt
...
 
+##or on a Mac
+ ```console
+ cd source/android
+ make
+ ```
+
 ## Build failure "UnsupportedClassVersionError 52.0"
 
 If you get a build failure similar to "UnsupportedClassVersionError 52.0" then from Android Studio go to: **File** -> **Other Settings** --> **Default project structure...** --> Check **Use Embedded JDK**
```
Comment (shalinijoshi19) [inline source/android/README.md]: were you able to get to the bottom of the long-filename related build failures? Should those be in the readme?

**Comment by shalinijoshi19** [comment]
What is the plan for unit tests?

File: source/android/README.md (hunk 1 above)
Comment (almedina-ms) [reply source/android/README.md]: By renaming the folder where the project is located I didn't had to face the issues again, the only caveat I found while making the sample app is that having our package contain Android in its name you end up having to write global::Android at some points but nothing else

Comment (shalinijoshi19) [reply source/android/README.md]: Might be a good idea to document the error/symptoms you were observing in some sort of "Troubleshooting" section 

---
In reply to: [348199717](https://github.com/microsoft/AdaptiveCards/pull/3610#discussion_r348199717) [](ancestors = 348199717)

**Comment by almedina-ms** [comment]
I was just recommended to use xunit or nunit for making unit tests, so I'll merge them in a different PR

---
In reply to: [555708172](https://github.com/microsoft/AdaptiveCards/pull/3610#issuecomment-555708172) [](ancestors = 555708172)

File: source/android/README.md (hunk 1 above)
Comment (almedina-ms) [reply source/android/README.md]: Should this be on the OneNote or in a different text file?

---
In reply to: [349356400](https://github.com/microsoft/AdaptiveCards/pull/3610#discussion_r349356400) [](ancestors = 349356400,348199717)

**Comment by paulcam206** [comment]
Closing for now -- we can rehydrate when we have cycles to pursue Xamarin support.

**Pull request closed without merging**